### ISSUE TITLE
Interface is down - private_network

### ISSUE THREAD

**Issue by karoldrazek** [mitchellh/vagrant#6976]
**VirtualBox:** 5.0.10 r104061
**Vagrant**: 1.8.1
**Plugins:**
vagrant-aws (0.6.0)
vagrant-berkshelf (4.1.0)
vagrant-hostmanager (1.6.1)
vagrant-omnibus (1.4.1)
vagrant-share (1.1.5, system)

Problematic Vagrantfile
```
VAGRANTFILE_API_VERSION = '2'

Vagrant.require_version '>= 1.5.0'

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  config.vm.hostname = 'aet-vagrant'
  config.vm.box = 'opscode-centos-6.7'
  config.vm.box_url = 'http://opscode-vm-bento.s3.amazonaws.com/vagrant/'\
    'virtualbox/opscode_centos-6.7_chef-provisionerless.box'

  config.vm.network :private_network, ip: '192.168.123.100'

  config.vm.synced_folder '/Users/karol/Repositories/local_folder',
    '/content/jetty',
    owner: 'root', group: 'root'

  config.vm.network 'forwarded_port', guest: 8181,  host: 8181

  config.vm.provider 'virtualbox' do |v|
    v.memory = 3072
    v.cpus = 2
  end

  config.omnibus.chef_version = 'latest'
  config.berkshelf.enabled = true

  config.vm.provision :chef_solo do |chef|
    chef.json = {
    }
    chef.run_list = [
    ]
  end

  config.vm.post_up_message = 'Banner message' 

end
```

After issuing '**vagrant reload**' the interface with 192.168.123.100 is not getting up during startup. I have to manually issue '**ifconfig eth1 up**' to get it up.

**Comment by sethvargo** [comment]
Hi @karoldrazek 

What is your host OS?

**Comment by karoldrazek** [comment]
Hello @sethvargo 

I've done some additional testing and narrowed down the problem a bit. I can replicate the issue 100% times. Please see PoC at https://github.com/karoldrazek/vagrant-issue-poc
Installation of group package 'Desktop' seems to be the cause.

Aside of all those versions mentioned above I bumped my VirtualBox to 5.0.14 r105127.
My OS: OSX El Capitan 10.11.3

Observations using given PoC:
1. After issuing initial `vagrant up` eth1 is up
2. After issuing `reboot` from within VM eth1 is up
3. After issuing `halt` from within VM and starting it in VBox console eth1 is up
4. After issuing `vagrant halt` and starting VM in Vbox console eth1 is up
5. After issuing `vagrant reload` eth1 is down
6. After issuing `vagrant halt` and then `vagrant up` eth1 is down
7. After issuing `vagrant halt` and then `vagrant up --no-provision` eth1 is down

**Comment by karoldrazek** [comment]
Small update:
Still doesn't work on Vagrant 1.8.4

**Comment by chrisroberts** [comment]
Should be fixed by #7751

**Comment by chrisroberts** [comment]
Needs repro with #7751 applied to ensure it properly addresses the issue.

**Comment by chrisroberts** [comment]
Fully fixed with #7926
